encode_test.go

```diff
@@ -112,10 +112,10 @@ var marshalTests = []struct {
 		map[string]any{"v": ""},
 		"v: \"\"\n",
 	}, {
-		map[string][]string{"v": []string{"A", "B"}},
+		map[string][]string{"v": {"A", "B"}},
 		"v:\n    - A\n    - B\n",
 	}, {
-		map[string][]string{"v": []string{"A", "B\nC"}},
+		map[string][]string{"v": {"A", "B\nC"}},
 		"v:\n    - A\n    - |-\n      B\n      C\n",
 	}, {
 		map[string][]any{"v": []any{"A", 1, map[string][]int{"B": []int{2, 3}}}},
```

Comment:
This might be what is failing in the CI

```suggestion
		map[string][]any{"v": []any{"A", 1, map[string][]int{"B": {2, 3}}}},
```